packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -953,11 +957,11 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
       removeBottom: true,
       removeLeft: true,
       removeRight: true,
-      child: Builder(
-        builder: (BuildContext context) {
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
           return CustomSingleChildLayout(
             delegate: _PopupMenuRouteLayout(
-              position,
+              positionBuilder?.call() ?? position!,
```

Comment:
Do we really need the `positionBuilder` parameter to fix this issue? It seems to me that with `LayoutBuilder` handling the rebuild, `position` is enough. The `positionBuilder` might be a useful feature for other uses, but unless we have explicit use case in mind (or brought up by an issue) it's unnecessary.

Reply:
I think the `positionBuilder` is necessary, When screen dimensions change (e.g. during rotation or window resizing), even with `LayoutBuilder`, the original position value remains static and may become incorrect relative to the new screen dimensions. The `positionBuilder` allows us to dynamically recalculate the position whenever the layout changes.

I think all about popup widget should need a positionBuilder replace position.

Reply:
I see. It is presented by `showMenu`, and therefore won't receive contextual updates any more.